DigitalLearningSolutions.Web/Controllers/SupervisorController/Supervisor.cs

```diff
@@ -60,6 +63,19 @@ public IActionResult Index()
             return View("MyStaffList", model);
         }
 
+        private DlsRole GetSupervisorDlsRole(SupervisorDelegateDetail supervisor)
+        {
+            var loggedInUserId = User.GetAdminId();
+            var loggedInAdminUser = userDataService.GetAdminUserById(loggedInUserId!.GetValueOrDefault());
+            bool activeAdminRecordAtSameCenter = loggedInAdminUser != null
+                && loggedInAdminUser.CentreId == supervisor.CentreId
+                && loggedInAdminUser.EmailAddress == supervisor.SupervisorEmail
+                && loggedInAdminUser.Active;
+            return activeAdminRecordAtSameCenter && loggedInAdminUser.IsSupervisor ? DlsRole.Supervisor

```

Comment:
This is getting the role (supervisor, nominated supervisor, learner) of the logged in user (the supervisor accessing the staff list interface) rather than each of the members of staff on the staff list. To get this, we will need to change the data service and model that retrieves the supervisorDelegateDetails to find matching AdminUser records for the delegate by outer joining Candidates to AdminUsers on email address, centre and active = 1 and then checking for the supervisor roles in the matched admin records.